app/src/main/java/pl/ludek/smat/home/pool_office_client/presentation/MainActivity.kt

```diff
@@ -0,0 +1,124 @@
+package pl.ludek.smat.home.pool_office_client.presentation
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import pl.ludek.smat.home.pool_office_client.R
+import pl.ludek.smat.home.pool_office_client.data.live_data.SensorsViewModel
+import pl.ludek.smat.home.pool_office_client.databinding.ActivityMainBinding
+import pl.ludek.smat.home.pool_office_client.domain.model.PoolInfoData
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var relayFirstOnStr:String    // "RELAY-FIRST-ON"
+    private lateinit var  relayFirstOffStr:String  // "RELAY-FIRST-OFF"
+    private lateinit var relayFiveOnStr:String    // "RELAY-FIVE-ON"
+    private lateinit var relayFiveOffStr:String  // "RELAY-FIVE-OFF"
+    private lateinit var relayAllOnStr:String    // "RELAY-ALL-ON"
+    private lateinit var relayAllOffStr:String   // "RELAY-ALL-OFF"
+    private lateinit var errorRelayStr:String
+    private lateinit var  errorSensorStr:String
+    private lateinit var dataFromRepository: PoolInfoViewModel
+    private lateinit var dataFromRelay: RelayViewModel
+    private val dataFromSensor: SensorsViewModel by viewModels()
```

Comment:
Modern Android apps uses MVVM architecture and usually you will see single `ViewModel` assigned to a given Activity.
In our case it would be the `MainActivityViewModel`, which should handle all the model related staff. Please read about the `Activity` life cycle on Android, you will notice that it can be unexpectedly closed (or even finalized) during the normal app use. That is why we shouldn't use it to keep any sensitive or heavy stuff. On the other hand, `ViewModel`s are kept in the memory all the time.
Summarizing, I wont you to move the logic related to those three classes into a single `MainActivityViewModel` class.